人智导论/ai-assignment1/code of new version framework/eightFigurePuzzles.cpp

```diff
@@ -0,0 +1,202 @@
+			return res;
+		}
+	}
+	return res;
+}
+
+// 更新puzzleNode的后继可操作动作状态，其中 （1，0）代表空格向上移动，（-1，0）代表空格向下移动，（0，1）代表空格向左移动，（0，-1）代表空格向右移动。
+PUZZLE_NODE updatePuzzleNodeActionList(PUZZLE_NODE puzzleNode)
+{
+	int *xyPosition = findZeroPosition(puzzleNode);
+	int x = xyPosition[0];
+	int y = xyPosition[1];
+	if (x >= 1)
+	{
+		vector<int> actionUp;
+		actionUp.push_back(1);
+		actionUp.push_back(0);
+		puzzleNode.nextActionList.push_back(actionUp);
+	}
+	if (x <= 1)
+	{
+		vector<int> actionDown;
+		actionDown.push_back(-1);
+		actionDown.push_back(0);
+		puzzleNode.nextActionList.push_back(actionDown);
+	}
+	if (y >= 1)
+	{
+		vector<int> actionLeft;
+		actionLeft.push_back(0);
+		actionLeft.push_back(1);
+		puzzleNode.nextActionList.push_back(actionLeft);
+	}
+	if (y <= 1)
+	{
+		vector<int> actionRight;
+		actionRight.push_back(0);
+		actionRight.push_back(-1);
+		puzzleNode.nextActionList.push_back(actionRight);
+	}
+	return puzzleNode;
+}
+
+void outputAction(vector<int> action, int index)
+{
+	/*cout << action[0] << " " << action[1] << endl;*/
+	if (action[0] == 1 && action[1] == 0)
+	{
+		cout << "步数 " << index << ":向上移动" << endl;
+		cout << endl;
+	}
+	else if (action[0] == -1 && action[1] == 0)
+	{
+		cout << "步数 " << index << "向下移动" << endl;
+		cout << endl;
+	}
+	else if (action[0] == 0 && action[1] == 1)
+	{
+		cout << "步数 " << index << "向左移动" << endl;
+		cout << endl;
+	}
+	else
+	{
+		cout << "步数 " << index << "向右移动" << endl;
+		cout << endl;
+	}
+}
+
+// 给定动作action（action为二维数组）和puzzleNode,返回执行该动作后新的节点
+PUZZLE_NODE moveToPuzzleNode(vector<int> action, PUZZLE_NODE puzzleNode)
+{
+	// cout << action[0] << " " << action[1] << endl;
+	// if (action[0] == 1 && action[1] == 0) {
+	//	cout << "向上移动" << endl;
+	// }
+	// else if (action[0] == -1 && action[1] == 0) {
+	//	cout << "向下移动" << endl;
+	// }
+	// else if (action[0] == 0 && action[1] == 1) {
+	//	cout << "向左移动" << endl;
+	// }
+	// else {
+	//	cout << "向右移动" << endl;
+	// }
+
+	int *xyPosition = findZeroPosition(puzzleNode);
+	int x = xyPosition[0];
+	int y = xyPosition[1];
+	PUZZLE_NODE nextPuzzleNode;
+	for (int xPos = 0; xPos < 3; xPos++)
+	{
+		for (int yPos = 0; yPos < 3; yPos++)
+		{
+			nextPuzzleNode.puzzle[xPos * 3 + yPos].xPosition = xPos;
+			nextPuzzleNode.puzzle[xPos * 3 + yPos].yPosition = yPos;
+			if (xPos == x && yPos == y)
+			{
+				nextPuzzleNode.puzzle[xPos * 3 + yPos].puzzleId = puzzleNode.puzzle[((x - action[0]) * 3 + (y - action[1]))].puzzleId;
+			}
+			else if (xPos == (x - action[0]) && yPos == (y - action[1]))
+			{
+				nextPuzzleNode.puzzle[xPos * 3 + yPos].puzzleId = puzzleNode.puzzle[(x * 3 + y)].puzzleId;
+			}
+			else
+			{
+				nextPuzzleNode.puzzle[xPos * 3 + yPos].puzzleId = puzzleNode.puzzle[xPos * 3 + yPos].puzzleId;
+			}
+		}
+	}
+	return nextPuzzleNode;
+}
+
+// 用于生成PuzzleNode中随机动作索引
+int getRandomNumber(int actionSize)
+{
+	int RandomNumber;
+	srand((unsigned)time(0));			// time()用系统时间初始化种。为rand()生成不同的随机种子。

```

Comment:
This is a definitely incorrect usage, use `srand` at the beginning of `main` instead, or replace it with [c++ style random number generation library](https://en.cppreference.com/w/cpp/numeric/random).